### PR TITLE
Add concurrency workers to chart/upgrade

### PR DESCRIPTION
**Pull request by rgee0**
## Description

Adds a worker pool (default 4) to chart/upgrade.go with a command line flag of `workers` or `-c` to change this value from the default.

## Motivation and Context

Charts with significant number of images were taking a long time to process in a single stream.

- [ ] I have raised an issue to propose this change, which has been given a label of `design/approved` by a maintainer ([required](https://github.com/alexellis/arkade/blob/master/CONTRIBUTING.md))

### Previously:
```
➜  faas-netes git:(workers) time make upgrade-charts                              
Upgrading images for all helm charts
2024/06/01 21:03:06 Wrote 9 updates to: ./chart/openfaas/values.yaml
2024/06/01 21:03:07 Wrote 1 updates to: ./chart/kafka-connector/values.yaml
2024/06/01 21:03:12 Wrote 1 updates to: ./chart/queue-worker/values.yaml
make upgrade-charts  1.38s user 1.27s system 16% cpu 16.148 total
```
### With 4 workers:
```
➜  faas-netes git:(testArkWorkers) time make upgrade-charts         
Upgrading images for all helm charts
2024/06/02 13:34:37 Wrote 7 updates to: ./chart/openfaas/values.yaml
2024/06/02 13:34:38 Wrote 1 updates to: ./chart/kafka-connector/values.yaml
2024/06/02 13:34:42 Wrote 1 updates to: ./chart/queue-worker/values.yaml
make upgrade-charts  1.34s user 1.22s system 31% cpu 8.137 total
```

### error output
```
➜  faas-netes git:(testArkWorkers) ✗ time make upgrade-charts                     
Upgrading images for all helm charts
Error: unable to list tags for ghcr.io/openfaasltd/autoscaler11
unable to list tags for ghcr.io/openfaasltd/faas-netes-oem6
unable to list tags for ghcr.io/openfaasltd/faas-netes4
unable to list tags for ghcr.io/openfaasltd/event-worker3
unable to list tags for ghcr.io/openfaasltd/openfaas-oidc-plugin13
unable to list tags for ghcr.io/openfaasltd/jetstream-queue-worker8
unable to list tags for ghcr.io/openfaas/gateway2
unable to list tags for ghcr.io/openfaasltd/gateway1
unable to list tags for ghcr.io/openfaasltd/openfaas-dashboard12
unable to list tags for ghcr.io/openfaas/faas-netes7
unable to list tags for ghcr.io/openfaasltd/queue-worker9
unable to list tags for ghcr.io/openfaasltd/faas-netes5
unable to list tags for ghcr.io/openfaas/ingress-operator14
unable to list tags for ghcr.io/openfaas/queue-worker10
make: *** [upgrade-charts] Error 1
make upgrade-charts  0.74s user 0.67s system 57% cpu 2.472 total
```

Note that this change also incorporates the smaller bug fix proposed in #1076, hence the change from 9 updates to 7 in `./chart/openfaas/values.yaml`

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Documentation

- [ ] I have updated the list of tools in README.md if (required) with `./arkade get --format markdown`
- [ ] I have updated the list of apps in README.md if (required) with `./arkade install --help`

## Checklist:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/alexellis/arkade/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`

- [ ] I have tested this on arm, or have added code to prevent deployment

### CODE REVIEW

**Comment by alexellis** [comment]
Thanks for this update. Can you show how errors look now when they're encountered during an update of images?

File: cmd/chart/upgrade.go [hunk 1]
```diff
@@ -83,38 +86,47 @@ Otherwise, it returns a non-zero exit code and the updated values.yaml file.`,
...
+				for image := range workChan {
+					if len(image) > 0 {
+						updated, imageNameAndTag, err := updateImages(image, verbose)
+						if err != nil {
+							errChan <- err
+							continue
+						}
+						if updated {
+							mu.Lock()
+							updatedCount++
```
Comment (alexellis) [inline cmd/chart/upgrade.go:110]: What are your thoughts on having a second map that starts out empty, and gets populated as images are updated? In that way, there'd be no need for updatedCount. I don't think you'd need to guard the destination map with a mutex either?

**Comment by rgee0** [comment]
> Thanks for this update. Can you show how errors look now when they're encountered during an update of images?

Added a view of the output to the PR body testing section

**Review by alexellis — APPROVED** [review]
Approved. Thanks for cutting the time in half.

**Comment by alexellis** [comment]
I don't think anything needs to be done for faasd/compose: https://github.com/openfaas/faasd/blob/master/Makefile#L78

**Pull request merged**